src/tasks/run_closed_loop.py

```diff
@@ -1,5 +1,54 @@
 """Run all components of the BCI closed loop."""
+import logging
+import os
 import subprocess
+import tempfile
+
+from neural_data_simulator.settings import LogLevel
+from neural_data_simulator.util.runtime import configure_logger
+from neural_data_simulator.util.runtime import initialize_logger
+
+SCRIPT_NAME = "nds-run-closed-loop"
+logger = logging.getLogger(__name__)
+
+# This is the pipe message indicating that the main task has finished
+MAIN_TASK_FINISHED_PIPE_MSG = "main_task_finished"
+
+
+def _terminate_process(label: str, popen_process: subprocess.Popen, timeout: int = 5):
+    logger.info(f"Terminating {label}")
+    popen_process.terminate()
+    try:
+        popen_process.wait(timeout)
+    except subprocess.TimeoutExpired:
+        logger.warning(f"{label} did not terminate in time. Killing it")
+        popen_process.kill()
+
+
+def _create_pipe_file(pipe_name: str) -> tuple[str, str]:
+    temp_dir = tempfile.mkdtemp()
+    pipe_path = os.path.join(temp_dir, pipe_name)
+    os.mkfifo(pipe_path)
+    return (temp_dir, pipe_path)
+
+
+def _wait_for_pipe_message(pipe_path: str, pipe_message: str) -> bool:
+    """Wait for a pipe message in a pipe file.
+
+    :param pipe_path: Path to the pipe file.
+    :param pipe_message:  Message to wait for.
+    :returns: True if the message was found, False if the process was interrupted.
+    """
```

Comment:
the docstrings should follow this standard:
https://github.com/agencyenterprise/neural-data-simulator/blob/main/src/neural_data_simulator/runner.py#L78